Common/SecurityIdentifier.cs

```diff
@@ -350,6 +350,35 @@ public SecurityIdentifier(string symbol, ulong properties, SecurityIdentifier un
         /// <returns>A new <see cref="SecurityIdentifier"/> representing the specified symbol today</returns>
         public static SecurityIdentifier GenerateEquity(string symbol, string market, bool mapSymbol = true, IMapFileProvider mapFileProvider = null)
         {
+            return GenerateWithFirstDate(symbol, market, SecurityType.Equity, mapSymbol, mapFileProvider);
+        }
+
+        /// <summary>
+        /// This implementation uses the configured <see cref="IMapFileProvider"/> via the <see cref="Composer.Instance"/>
+        /// </summary>
+        /// <param name="symbol">The symbol as it is known today</param>
+        /// <param name="market">The market</param>
+        /// <param name="securityType">The security type of the symbol</param>
+        /// <param name="mapSymbol">Specifies if symbol should be mapped using map file provider</param>
+        /// <param name="mapFileProvider">Specifies the IMapFileProvider to use for resolving symbols, specify null to load from Composer</param>
+        /// <returns>A new <see cref="SecurityIdentifier"/> representing the specified symbol today</returns>
+        public static SecurityIdentifier GenerateWithFirstDate(string symbol, string market, SecurityType securityType, bool mapSymbol = true, IMapFileProvider mapFileProvider = null)
```

Comment:
We might want to make this private. I view this 'first date' behavior as an internal implementation detail of the `SecurityIdentifier` and as such, shouldn't be publicly exposed unless absolutely necessary.

Reply:
Yea, in general you want to test a type's public API so that it's internals can change w/out having to update everyone else. If you find yourself _needing_ to test internal logic, it's usually a [code smell](https://en.wikipedia.org/wiki/Code_smell) indicating that the type is too complicated and some of the internal logic that requires testing can be extracted into it's own type complete with it's own public API.